src/librustc_mir/borrow_check/places_conflict.rs

```diff
@@ -328,14 +328,18 @@ fn place_element_conflict<'a, 'gcx: 'tcx, 'tcx>(
             }
         }
         (Place::Promoted(p1), Place::Promoted(p2)) => {
-            if p1.0 == p2.0 {
-                // the same promoted - base case, equal
-                debug!("place_element_conflict: DISJOINT-OR-EQ-PROMOTED");
-                Overlap::EqualOrDisjoint
-            } else {
+            if p1.0 != p2.0 {
                 // different promoteds - base case, disjoint
                 debug!("place_element_conflict: DISJOINT-PROMOTED");
                 Overlap::Disjoint
+            } else if mir.promoted[p1.0].ignore_borrow_conflicts {
```

Comment:
couldn't we just check for zero length array type here instead of infecting everything with that info?

Reply:
i'm with @oli-obk here: Lets see if that (presumably simpler) fix does the job here?

Reply:
I think I was concerned about performance thinking that it would need to be check way more than it actually is (which is once per promoted).